### PR TITLE
177602476: created reminder notes UI changes with tests and modified db schema

### PR DESCRIPTION
**Pull request by JustinRWong**
This PR achieves the UI change for the Reminder Email Notes. These are the changes in this PR
- **app/controllers/options_controller.rb**: rebased with main.

- **app/views/mailer/confirm_order.html.erb**: includes user input sanitation for the reminder notes input.

- **app/views/mailer/confirm_reservation.html.erb**: actual UI change in the options route.

- **app/views/options/index.html.haml**: main point of change was "general_reminder_email_notes" on line 134. This renders the UI change.

- **config/locales/en.option_descriptions.yml**: changed to account for help box on hover.

- **db/migrate/20210405142646_add_reminder_emails_to_option.rb**:  db migration change that modifies options schema to account for reminder email notes.

- **db/schema.rb**: modfied after running rake db:migrate. This should be included if we do not want to run "heroku run rake db:migrate". 

- **features/season_setup/change_options.feature**: cucumber test to ensure that the "General Reminder Email Notes" actually gets rendered on the UI.

Here is a screenshot of the UI change in the options route. Specifically, the difference is the "General Reminder Email Notes"

Screenshot 2021-04-10 at 8 01 24 PM

### CODE REVIEW

**Comment by armandofox** [comment]
There is a structural problem here.  The reminder email needs to actually be a separate view from the regular order-confirmation email; what you've done is incorporate the reminder text into an order confirmation, rather than establishing a way to send another email whose content is the reminder.  Does that make sense?  So I would expect to see a new view in `app/views/mailer` that will be used when sending a reminder email, and when we eventually get Mailchimp campaign integration, that view would be instantiated but rather than being delivered via ActionMailer it would become part of some API call to set the content of a campaign.  But the way you have it now, every email that confirms any type of order (which includes purchases, refunds, donations, comp tickets, etc) will incorporate the "reminder email" text, when in fact that text should only be used in reminder emails which are separate.

File: app/views/mailer/confirm_reservation.html.erb [hunk 1]
```diff
@@ -17,7 +17,13 @@
     <%= Option.general_confirmation_email_notes %>
   </p>
 
+  <p class="general-reminder-email-notes">
+    <%= Option.general_reminder_email_notes %>
```
Comment (abhinavDhulipala) [inline app/views/mailer/confirm_reservation.html.erb]: Maybe pull this out to a Controller variable?

**Review by abhinavDhulipala — COMMENTED** [review]
Can probably remove ```app/controllers/options_controller.rb``` since this only has a spacing change

File: config/locales/en.option_descriptions.yml [hunk 2]
```diff
@@ -327,12 +327,12 @@ en:
...
     regular_order_service_charge: >
 
-      A flat fee (service charge) assessed on regular (non-subscription) orders.  
+      A flat fee (service charge) assessed on regular (non-subscription) orders.
       Leave blank or enter 0 for no charge.
 
     regular_order_service_charge_description: >
 
-      A description that will appear on the order summary for regular 
+      A description that will appear on the order summary for regular
```
Comment (mattfyke) [inline config/locales/en.option_descriptions.yml:335]: interesting how there are many places where whitespaces were removed, seems like a linter doing its job.

File: features/season_setup/change_options.feature [hunk 3]
```diff
@@ -32,3 +32,16 @@ Scenario: invalid HTML email template because no placeholder
...
+  And I visit the admin:settings page
+  Then I should see "General Reminder Email Notes"
+  When I fill in "General Reminder Email Notes" with "This is an email reminder"
+  And I press "Update Settings"
+  Then the setting "General Reminder Email Notes" should be "This is an email reminder"
+
+Scenario: reminder emails feature iss disabled
+  When I disable the reminder email feature
+  And I visit the admin:settings page
+  Then I should not see "General Reminder Email Notes"
```
Comment (mattfyke) [inline features/season_setup/change_options.feature:47]: Good test added for the new functionality.

**Review by mattfyke — APPROVED** [review]
Overall a pretty simple PR that does one thing. Good job keeping this PR somewhat small. Also, good job using a migration to change the schema. LGTM

**Pull request merged**